### DISCUSSION TITLE
TreeView Not Rendering Correctly

### DISCUSSION THREAD

**Discussion by CHIPSNetwork** [enisn/UraniumUI#565] [General]
I'm following the docs as closely as possible.   I fail to get the icon/character/symbol to the left of an unexpanded group.
What's the trick to getting this to appear?

My contentpage:

<?xml version="1.0" encoding="utf-8" ?>
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:uranium="http://schemas.enisn-projects.io/dotnet/maui/uraniumui"
             xmlns:material="http://schemas.enisn-projects.io/dotnet/maui/uraniumui/material"
             xmlns:fa="clr-namespace:UraniumUI.Icons.FontAwesome;assembly=UraniumUI.Icons.FontAwesome"
             xmlns:m="clr-namespace:UraniumUI.Icons.MaterialIcons;assembly=UraniumUI.Icons.MaterialIcons"
             x:Class="ChipsAppV3.NotesPage"
             Title="NotesPage">
    
    <ContentPage.ToolbarItems>
        <ToolbarItem Text="Add New"  />
    </ContentPage.ToolbarItems>
    <ContentPage.Content>
        <ScrollView>
            <StackLayout>
                <material:TreeView x:Name="treeView" ItemsSource="{Binding Nodes}" ChildrenBinding="{Binding Children}" SelectionMode="Single" HeightRequest="300" WidthRequest="300" >
                   
                </material:TreeView> 
                <SearchBar x:Name="CustomerSearch"  Placeholder="Select Customer..." TextChanged="OnTextChanged" ></SearchBar>
                <ListView x:Name="searchResults"  HorizontalOptions="CenterAndExpand" ItemTapped="OnItemTapped" HeightRequest="500" MinimumHeightRequest="400" IsVisible="False">
                    <ListView.ItemTemplate>

                        <DataTemplate>
                            <ViewCell>
                                <Grid>
                                    <Label Grid.Column="0" Text="{Binding CustName}" />
                                </Grid>
                            </ViewCell>
                        </DataTemplate>
                    </ListView.ItemTemplate>
                </ListView>
            
            </StackLayout>
        </ScrollView>
    </ContentPage.Content>

</ContentPage>

My Mauiprogram.cs:

![image](https://github.com/enisn/UraniumUI/assets/59516001/5dcd7927-0bb1-4d8f-962d-cc641780c545)

**Comment by CHIPSNetwork** [comment]
My .cs file for the page:

using ChipsAppV3.Models;
using Microsoft.Maui.Controls;
using System.Collections.ObjectModel;

namespace ChipsAppV3;

public partial class NotesPage : ContentPage
{
    int New_Customer_ID = 0;
    public ObservableCollection<MyItem> Nodes { get; set; } = new();
    public NotesPage()
    {
        AddNodes();
        BindingContext = this;

        InitializeComponent();

        
    }

    
    void OnTextChanged(object sender, TextChangedEventArgs e)
    {
        SearchBar searchBar = (SearchBar)sender;
        var results = GlobalStuff.GlobalCustomer_Brief_ViewModel.GetSearchResults(searchBar.Text);
        searchResults.ItemsSource = results;
        searchResults.IsVisible = true;
        if (e.NewTextValue == string.Empty || e.NewTextValue == null)
        {
            searchResults.IsVisible = false;
        }
    }

    public async void OnItemTapped(object o, ItemTappedEventArgs e)
    {
        Customer_Brief_Item dataItem = (Customer_Brief_Item)e.Item;

        CustomerSearch.Text = "";
        CustomerSearch.Placeholder = dataItem.CustName;
        New_Customer_ID = dataItem.CustID;
        OnPropertyChanged(nameof(CustomerSearch.Placeholder));
        searchResults.IsVisible = false;
        await reload_notes();
    }

    async Task reload_notes()
    {
    }

   

      

        public void AddNodes()
        {
            Nodes.Add(new MyItem("A")
            {
                Children =
            {
                new MyItem("A.1"),
                new MyItem("A.2"),
            }
            });
            Nodes.Add(new MyItem("B")
            {
                Children =
            {
                new MyItem("B.1")
                {
                    Children =
                    {
                        new MyItem("B.1.a"),
                        new MyItem("B.1.b"),
                        new MyItem("B.1.c"),
                        new MyItem("B.1.d"),

                    }
                },
                new MyItem("B.2"),
            }
            });
            Nodes.Add(new MyItem("C"));
            Nodes.Add(new MyItem("D"));
        }
    
    public class MyItem
    {
        
        public MyItem(string name) // For easy initialization (optional)
        {
            Name = name;
        }

        public virtual string Name { get; set; }
        public virtual IList<MyItem> Children { get; set; } = new ObservableCollection<MyItem>();
    }

}

**Reply by CHIPSNetwork** [reply]
Version 2.7.4

**Comment by jiiteel** [comment]
I was able to get the arrows by modifying the App.xaml.
First add the namespace 

    xmlns:material="clr-namespace:UraniumUI.Material.Resources;assembly=UraniumUI.Material"

and then replace the line

  `<ResourceDictionary Source="Resources/Styles/Styles.xaml" />`

 With

   ```
<ResourceDictionary x:Name="appStyles" Source="Resources/Styles/Styles.xaml" />
   <material:StyleResource BasedOn="{x:Reference appStyles}" />
```

The opening  and closing of the nodes do not work however. So the open/closed state stays fixed to the way the node was created.

**Reply by jiiteel** [reply]
After the Uranium UI doc pages are up, I found the way to get this working in "Getting Started"
https://enisn-projects.io/docs/en/uranium/latest/Getting-Started
